py3status/modules/imap.py

```diff
@@ -35,63 +33,69 @@ class Py3status:
     criterion = 'UNSEEN'
     format = 'Mail: {unseen}'
     hide_if_zero = False
-    imap_server = '<IMAP_SERVER>'
     mailbox = 'INBOX'
-    new_mail_color = ''
-    password = '<PASSWORD>'
+    password = None
     port = '993'
     security = 'ssl'
-    user = '<USERNAME>'
+    server = None
+    user = None
+
+    class Meta:
+        deprecated = {
+            'rename': [
+                {
+                    'param': 'new_mail_color',
+                    'new': 'color_good',
+                    'msg': 'obsolete parameter use `color_good`',
+                },
+                {
+                    'param': 'imap_server',
+                    'new': 'server',
+                    'msg': 'obsolete parameter use `server`',
+                },
+            ],
+        }
 
     def post_config_hook(self):
         if self.security not in ["ssl", "starttls"]:
             raise ValueError("Unknown security protocol")
 
-    def check_mail(self):
+    def imap(self):
         mail_count = self._get_mail_count()
+        color = self.py3.COLOR_BAD
+        full_text = STRING_UNAVAILABLE
 
-        response = {'cached_until': self.py3.time_in(self.cache_timeout)}
-
-        if not self.new_mail_color:
-            self.new_mail_color = self.py3.COLOR_GOOD
-
-        if mail_count == 'N/A':
-            response['full_text'] = mail_count
-        elif mail_count != 0:
-            response['color'] = self.new_mail_color
-            response['full_text'] = self.py3.safe_format(
-                self.format, {'unseen': mail_count}
-            )
+        if mail_count == 0 and self.hide_if_zero:
+            full_text = ''
         else:
-            if self.hide_if_zero:
-                response['full_text'] = ''
-            else:
-                response['full_text'] = self.py3.safe_format(
-                    self.format, {'unseen': mail_count}
-                )
+            color = self.py3.COLOR_GOOD
+            full_text = self.py3.safe_format(self.format, {'unseen': mail_count})
```

Comment:
I think you are missing the `mail_count is None` case.  Plus in that case I think we want to use the format still but {mail_count} is STRING_UNAVAILABLE = 'N/A'